test/functional/api/server_requests_spec.lua

```diff
@@ -137,6 +137,12 @@ describe('server -> client', function()
   end)
 
   describe('when the client is a recursive vim instance', function()
```

Comment:
Any of the tests in here may hang the OSX build. So it may be related to the `before_each` or `after_each` steps.